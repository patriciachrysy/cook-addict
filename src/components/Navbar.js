import React from 'react';
import OverlayTrigger from 'react-bootstrap/OverlayTrigger';
import Popover from 'react-bootstrap/Popover';
import { NavLink, useNavigate, useLocation } from 'react-router-dom';
import { GoHome } from 'react-icons/go';
import { BiSearchAlt } from 'react-icons/bi';
import { GiCook } from 'react-icons/gi';
import { IoIosArrowBack } from 'react-icons/io';
import Filter from './Filter';
import '../styles/Navbar.css';

const popover = (
  <Popover id="popover-basic" className="filter-popover">
    <Popover.Header as="h4">Find recipes</Popover.Header>
    <Popover.Body>
      <Filter />
    </Popover.Body>
  </Popover>
);

const Navbar = () => {
  const navigate = useNavigate();
  const location = useLocation();

  const goBack = (e) => {
    e.preventDefault();
    navigate(-1);
  };

  return (
    <nav>
      <div>
        { location.pathname !== '/'
          && (
          <button type="button" onClick={(e) => goBack(e)}>
            <IoIosArrowBack />
          </button>
          )}
        <NavLink exact className="navbar__link" to="/">
          <GoHome />
        </NavLink>
      </div>
      <span>Cook Addict</span>
      <div>
        <NavLink exact className="navbar__link" to="/recipes/all">
          <GiCook />
        </NavLink>
        <OverlayTrigger trigger="click" placement="bottom" overlay={popover} rootClose>
          <button type="button">
            <BiSearchAlt />
          </button>
        </OverlayTrigger>
      </div>
    </nav>
  );
};

export default Navbar;
