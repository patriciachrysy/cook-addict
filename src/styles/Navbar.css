@import url('https://fonts.googleapis.com/css2?family=Pacifico&display=swap');

nav {
    width: 100%;
    display: flex;
    justify-content: space-around;
    padding: 0 5px;
    align-items: center;
    height: 10vh;
}

nav svg {
    font-size: 25px;
    color: black;
}

nav span {
    width: 70%;
    text-align: center;
    font-family: 'Pacifico', cursive;
}

nav div {
    display: flex;
    align-items: center;
}

nav button {
    margin-left: 10px;
    background: none;
    border: none;
    font-size: 25px;
}