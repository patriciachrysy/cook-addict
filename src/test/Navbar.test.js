import React from 'react';
import renderer from 'react-test-renderer';

import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import Navbar from '../components/Navbar';

import store from '../Redux/ConfigureStore';

it('renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <Navbar />
      </Router>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render Navbar component', () => {
  render(<Provider store={store}><Router><Navbar /></Router></Provider>);
  screen.debug();
});
