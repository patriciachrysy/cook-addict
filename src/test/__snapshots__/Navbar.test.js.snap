// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav>
  <div>
    <a
      aria-current="page"
      className="navbar__link active"
      exact={true}
      href="/"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 9l-3-3V2h-2v2L8 1 0 9h2l1 5c0 .55.45 1 1 1h8c.55 0 1-.45 1-1l1-5h2zm-4 5H9v-4H7v4H4L2.81 7.69 8 2.5l5.19 5.19L12 14z"
          fillRule="evenodd"
        />
      </svg>
    </a>
  </div>
  <span>
    Cook Addict
  </span>
  <div>
    <a
      className="navbar__link"
      exact={true}
      href="/recipes/all"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M255 22.31c-21.2 0-40.5 10.78-50.5 27.63l-4.4 7.42-7.6-4.08c-7.1-3.85-15.2-5.89-23.3-5.89-28.3 0-51.3 23.57-51.3 53.21 0 26.5 18.6 48.6 43.5 52.6l7.6 1.2v53.4c57.9-11.1 116-11.6 174 0v-69.7l4.8-2.6c15.7-8.3 26.1-26.1 26.1-46.15 0-28.8-20.7-51.12-45.4-51.12-6.5 0-13 1.61-18.9 4.72l-6.2 3.2-4.6-5.09C288 29.29 272 22.31 255 22.31zm1 194.79c-25.6 0-51.1 2.4-76.7 7-.9 6-1.3 12.2-1.3 18.6 0 29 9.2 55.2 23.6 73.7 14.4 18.5 33.5 29.3 54.4 29.3s40-10.8 54.4-29.3c14.4-18.5 23.6-44.7 23.6-73.7 0-6.4-.5-12.7-1.3-18.7-25.6-4.7-51.1-7-76.7-6.9zM208.7 348l-89.2 29.7-27.98 112H321.6L307.1 388l17.8-2.6 14.9 104.3h80.7l-28-112-89.2-29.7c-13.8 9.9-30 15.7-47.3 15.7-17.3 0-33.5-5.8-47.3-15.7zm69.3 52.7a9.999 9.999 0 0 1 10 10 9.999 9.999 0 0 1-10 10 9.999 9.999 0 0 1-10-10 9.999 9.999 0 0 1 10-10zm7 46a10 10 0 0 1 10 10 10 10 0 0 1-10 10 10 10 0 0 1-10-10 10 10 0 0 1 10-10z"
        />
      </svg>
    </a>
    <button
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10 18a7.952 7.952 0 0 0 4.897-1.688l4.396 4.396 1.414-1.414-4.396-4.396A7.952 7.952 0 0 0 18 10c0-4.411-3.589-8-8-8s-8 3.589-8 8 3.589 8 8 8zm0-14c3.309 0 6 2.691 6 6s-2.691 6-6 6-6-2.691-6-6 2.691-6 6-6z"
        />
        <path
          d="M11.412 8.586c.379.38.588.882.588 1.414h2a3.977 3.977 0 0 0-1.174-2.828c-1.514-1.512-4.139-1.512-5.652 0l1.412 1.416c.76-.758 2.07-.756 2.826-.002z"
        />
      </svg>
    </button>
  </div>
</nav>
`;
